vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -358,8 +359,19 @@ func skipIPAddrForWaiter(ip net.IP, ignoredGuestIPs []interface{}) bool {
 		return true
 	default:
 		for _, ignoredGuestIP := range ignoredGuestIPs {
-			if ip.String() == ignoredGuestIP.(string) {
-				return true
+			if strings.Contains(ignoredGuestIP.(string), "/") {
+				_, ignoredIPNet, err := net.ParseCIDR(ignoredGuestIP.(string))
+				if err != nil {
+					log.Printf("[ERROR] Failed to parse ignored_guest_ips entry: %q", ignoredGuestIP)
+					continue
```

Comment:
I definitely agree that checking if the string actually represents a CIDR is the right thing to do, but in this case just dropping it and moving to the next element in the list may cause trouble, since all that happens during runtime.

How about moving the checks into a validation function in the schema? What I was thinking is that you could replace resource_vsphere_virtual_machine.go:116 with something like this:

```
			Elem:        &schema.Schema{
				Type: schema.TypeString,
				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
					v := val.(string)
                                        // cidr and ip validatio logic goes here
					return
				},
			},

```

An example of a simple validation function can be found here:
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#validatefunc

Since the parameters will be validated when terraform evaluates the config we can be confident that when skipIPAddrForWaiter is called the list will only contain either IPs or CIDRs.

Reply:
Great idea. Implemented in 6d76080.